azure-active-directory-b2c/sample-06-resource-server-validate-audience/src/main/java/com/spring/sample/b2c/security/WebSecurityConfiguration.java

```diff
@@ -0,0 +1,42 @@
+package com.spring.sample.b2c.security;
+
+import java.util.ArrayList;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.oauth2.core.DelegatingOAuth2TokenValidator;
+import org.springframework.security.oauth2.jwt.JwtClaimNames;
+import org.springframework.security.oauth2.jwt.JwtClaimValidator;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder;
+
+/**
+ * Security Configuration.
+ */
+@EnableWebSecurity
+public class WebSecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+  @Value("${spring.security.oauth2.resourceserver.jwt.jwk-set-uri}")
+  String jwkSetUri;
+
+  @Value("${validateAudience}")
+  String validateAudience;
+
+  @Override
+  protected void configure(HttpSecurity http) throws Exception {
+    http.authorizeRequests((requests) -> requests.anyRequest().authenticated())
+        .oauth2ResourceServer()
+        .jwt();
+  }
+
+  @Bean
+  JwtDecoder jwtDecoder() {
+    NimbusJwtDecoder jwtDecoder = NimbusJwtDecoder.withJwkSetUri(this.jwkSetUri).build();
+    jwtDecoder.setJwtValidator(new DelegatingOAuth2TokenValidator<>(
+        new JwtClaimValidator(JwtClaimNames.AUD, aud ->
+          aud != null && ((ArrayList) aud).contains(this.validateAudience))));
```

Comment:
`aud` cliam value is String, not ArrayList. 

Refs: https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens#payload-claims